src/TuxxState.js

```diff
@@ -0,0 +1,340 @@
+// @param input UNKNOWN: value we are performing invariant check upon
+var invariantValueCheck = function (input) {
+  if (typeof(input) === "object") {
+    invariant(!input, 'Cannot perform operation on "%s" because it must not be an array or object.', input);
+  }
+};
+
+// invariantNumberOrStringCheck FUNCTION: internal function that ensures a given value is a number or string or throws an error
+// @param input UNKNOWN: value we are performing invariant check upon
+var invariantNumberOrStringCheck = function (input) {
+  if (typeof(input) !== "number" && typeof(input) !== "string") {
+    invariant(!input, 'Cannot perform operation on "%s" because it is not of type number or of type string.', input);
+  };
+};
+
+// invariantArgCheck FUNCTION: internal function that ensures a given value is defined and is an object or throws an error
+// @param input UNKNOWN: value we are performing invariant check upon
+var invariantArgCheck = function (input) {
+  if (!input) {
+    invariant(input, 'This function requires an object as an argument.');
+  } else if (Object.prototype.toString.call(input) !== '[object Object]') {
+    invariant(!input, 'This function requires an object as an argument.');
+  }
+};
+
+// mixin to React class that will provide convenience methods for updating state
+module.exports = {
+  // addState FUNCTION: adds the values of the deepest keys in the passed in object to the corresponding deepest keys in the current state, or throws an error if keys don't match
+  // @param propsToAdd OBJECT: required object argument where the deepest keys are numbers or strings
+  // @param callback FUNCTION: optional callback argument that will be executed once setState is completed and the component is re-rendered
+  addState: function (propsToAdd, callback) {
+    // throw error if propsToAdd is undefined or not an object
+    invariantArgCheck(propsToAdd);
+    // build new state object
+    var newState = buildNewState(this.state, propsToAdd, function (newState, key, currentStateAtKey, newPropsAtKey) {
+      // throw error if newPropsAtKey is not a number or string
+      invariantNumberOrStringCheck(newPropsAtKey);
+      // perform addition on the corresponding deep keys
+      newState[key] = currentStateAtKey + newPropsAtKey;
+    });
+    // update state with new values and pass callback (if provided), triggering re-render
+    this.setState(newState, callback);
+  },
+
+  // subtractState FUNCTION: subtracts the values of the deepest keys in the passed in object from the corresponding deepest keys in the current state, or throws an error if keys don't match
+  // @param propsToSubtract OBJECT: required object argument where the deepest keys are numbers
+  // @param callback FUNCTION: optional callback argument that will be executed once setState is completed and the component is re-rendered
+  subtractState: function(propsToSubtract, callback) {
```

Comment:
The contents of add, subtract, multiply, and divide state all seem relatively redundant. Is there a way to DRY up the code?